src/types.ts

```diff
@@ -57,13 +62,20 @@ export interface ISymbolTableVisitor {
     scope: c3.ScopedSymbol;
 }
 
-export type GetParseTree<P> = (parser: P, type?: TableQueryPosition['type']) => ParseTree;
+export type GetParseTree<P> = (
+    parser: P,
+    type?: TableQueryPosition['type'] | 'select',
+) => ParseTree;
 
+export type GenerateSuggestionsFromRulesReturnType = Partial<AutocompleteParseResult> & {
+    shouldSuggestColumns?: boolean;
+    shouldSuggestColumnAliases?: boolean;
+};
```

Comment:
Let's rename it to `GenerateSuggestionsFromRulesResult`? It will be consistent with `AutocompleteParseResult`